lib/dispatcher/agent.js

```diff
@@ -110,6 +110,16 @@ class Agent extends DispatcherBase {
 
     await Promise.all(destroyPromises)
   }
+
+  get stats () {
+    const allClientStats = []
+    for (const client of this[kClients].values()) {
+      if (client.stats) {
+        allClientStats.push([client[kUrl].origin, client.stats])
+      }
+    }
```

Comment:
instead of an array of arrays, I would use an object.

Reply:
Sure thing. I assume [this]([52e7d94](https://github.com/nodejs/undici/pull/4157/commits/52e7d940bf3d0669a93b50d5f9fe6954614df57d)) is what you had in mind?